### ISSUE TITLE
Unholy screech while trying to play file with DTS:X audio.

### ISSUE THREAD

**Issue by Tyestor** [UniversalMediaServer/UniversalMediaServer#1258]
Here's the mediainfo: https://pastebin.com/gMuZhAg0

If you need a log, feel free to ask.

**Comment by Nadahar** [comment]
We always need a (trace) log, without it we're basicly clueless as to what's happening and can only speculate.

**Comment by Tyestor** [comment]
@Nadahar Yeah I actually had some time now and just got some logs for you.

[ums_dbg.zip](https://github.com/UniversalMediaServer/UniversalMediaServer/files/908858/ums_dbg.zip)

**Comment by Sami32** [comment]
The first thing that i saw without go deeper, is that we have a sample frequency parsing issue:

- From your MediaInfo log: `Sampling rate                            : 48.0 KHz`
- From your log: `sample frequency:null`

> Matched audio track: id: 0, lang: en, audio track title from metadata: DTS:X/DTS-HD Master Audio / 7.1 / 48 kHz / 7388 kbps / 24-bit, audio codec: DTS, sample frequency:null, number of channels: 8, bits per sample: 16

**Comment by Tyestor** [comment]
So what does that mean and what can I do about it?

**Comment by Nadahar** [comment]
@Tyestor I means that we probably have a bug to fix. What would be nice is if we could get our hands on a sample file. If you can't make it available public, you can send a PM on the [forum](http://www.universalmediaserver.com/forum/), although it's easier if we all have access to a sample. The sample must reproduce the problem obviously.

**Comment by Sami32** [comment]
@Tyestor You can cut a little part of this video file and post it here or PM it to the forum.
http://www.chrysocome.net/dd
https://sourceforge.net/projects/windd/files/windd/WinDD%20Release%201.0/WinDD.zip/download

Just download  this very tiny utility tool called _DD for Windows_, and type the following command line:
`dd if=sample-file of=small-sample-file bs=1024 count=10000`

**Note**: Don't forget to play the sample and verify that your issue still occur.

**Comment by Tyestor** [comment]
@Nadahar @Sami32 

Hey guys, sorry I've been really busy. The program/command you gave me doesn't work, Sami. It just opens WinDD and does nothing.

I've also had some other recurring problems for a long time, probably not related. I get the error 'This content cannot be played (800288e1)'. Most of these files are VC-1/AVC and have DTS-HD/DTS-DA MA encodes (both 5.1 and 7.1). I also get 'Unsupported Data' occasionally in the Transcode folder for a number for files but I rarely (would like to make it never) have to go in there. This isn't that big of an issue. I'll upload logs of some files I'm having trouble with and also link you to them on the forums.

Here are my audio settings for the below logs.

UMS: https://i.imgur.com/axfKLC4.png - Normally, I have everything unchecked, save for LPCM (but only sometimes). In this case though, it doesn't matter what I have checked.
PS3:  https://imgur.com/gallery/f0ttp - Everything not shown is unchecked.

[This content cannot be played - 1.zip](https://github.com/UniversalMediaServer/UniversalMediaServer/files/980715/1.zip)

[This content cannot be played - 2.zip](https://github.com/UniversalMediaServer/UniversalMediaServer/files/980716/2.zip)

And for good measure,

[Unholy Screech - 3.zip](https://github.com/UniversalMediaServer/UniversalMediaServer/files/980717/3.zip)

Pressing select would show 'LPCM' while playing Unholy Screech, but if I unchecked it in UMS, it would still use LPCM. The video works fine, audio doesn't, for the record. A lot of other things just show 'Dolby Digital'.

Anyway, sorry I couldn't be of more help.

**Comment by Nadahar** [comment]
@Tyestor 

FYI: We get all your configuration settings in the "packed debug files", so you don't have to take screenshots of that. We need sample files of content that won't play in most cases (unless the log file reveals an obvious cause). Error messages given by the renderer/PS3 often isn't very helpful to us, as we have noe way of knowing what they actually mean. When you give them, make sure to note that this is from the renderer and not UMS - to avoid confusion. I don't think the output settings in the PS3 has any relevance, I assume you use the PS3 as the renderer (the party which is communicating with UMS) and as such we don't really care what the PS3 does with the content after it has received it. The focus is on delivering the content from UMS to the PS3.

I haven't tested WinDD, but it's a very simply program so I'm pretty confident that it does work. I don't know how computer savvy you are, but it sounds like you've attempted to just doubleclick it (run it like a GUI application). That won't do, as it's a command-line application. Open the command line by running "cmd". Place WinDD and the file you want to process in the same folder, go there using ```cd``` in the command line window and then run the WinDD command @Sami32 explained.

**Comment by Sami32** [comment]
@Tyestor Could post a copy of the DD command line you typed ?

Sorry, i was not too explicit on how to use it, hopefully @Nadahar did it, and he is good at that ;-)
The main advantage of this free tiny tool, is that it do not alterate the properties of the media file, like most of the other tool do.

`sample frequency:null` same same, without sample it hard to know what's happen, because your MediaInfo log did not give us anything that could explain why, AFAIK.

Not that it will affect your issue, but you should set `tsmuxer_forcefps = true` in your _ums.conf file_; you can also do that from the GUI in the tsMuXeR video engine parameters while selecting "use FPS from FFmpeg" or something like that.

@SubJunk is the PS3 most experimented guy here, and one of the few having one as well.

**Comment by Tyestor** [comment]
@Nadahar - I already did all that. I opened an elevated command prompt and extracted WinDD into the movie folder, then changed the directory to that folder. I then ran the command that @Sami32 posted, making sure to change the 'if' and 'of' parameters. Error: 'dd' is not a recognized internal or external command, operable program or batch file. That's probably because the .exe is named 'WinDD' not 'dd'. So I changed it to WinDD. All that command does is open the program, it doesn't create a sample file.

UMS > PS3 > TV, over ethernet/hdmi. 

Thanks Sami, I'll do that.

**Comment by Nadahar** [comment]
@Tyestor I'm sorry, I've never tried WinDD and assumed that @Sami32 had verified that it actually behaved like ```dd```. After reading it's documentation I see that it's nothing like ```dd``` and can't do what we need in this case. All it can do is to "dump"/copy disk partitions, which is irrelevant in this context.

**Comment by Nadahar** [comment]
@Tyestor It seems like [partcopy](http://www.virtualobjectives.com.au/utilitiesprogs/partcopy.htm) should be able to do it. The goal is simply to copy the start of one of the problematic files. Make sure to copy from the very start and some megabytes ahead. There's a limit on 10 MB to attach files here, so to make it simple make sure to take less than 10 MB's,

**Comment by Sami32** [comment]
I always use the first link (DD) without any issue. I never used the second link tough.

**Comment by Nadahar** [comment]
@Sami32 I've told you that you have to describe your links 😉 I didn't even see that there were two links.

**Comment by Tyestor** [comment]
@Sami32 - You're right, I didn't try the first link, I tried the second link. Probably because I recognised Sourceforge and I just assumed the first link was a linux binary because the second link had 'WinDD' in it. 

@Nadahar I'll try that program if dd doesn't work for me.

Thanks guys.

**Comment by Tyestor** [comment]
Okay, the command/program works now using the first link. Slight issue (or non-issue), the file plays for 5 seconds and then stops, is that the expected behaviour?

**Comment by Nadahar** [comment]
@Tyestor That is the expected behavior, you have cut of just 1 MB from the start of the file. It might be easier to troubleshoot if we have a little more, could you try with 4096 instead of 1024?

In any case, what's important is that you confirm that the partial file has the same issue with UMS as the original. If so, please upload the small file here so that we can use it for troubleshooting the issue.

Just rename the file so that you add ".txt" to the filename (ignore Windows' warning that this might do terrible things) and drag and drop it into your comment window here. Wait for the file to finish upload, and post the comment.

As said above, I'd prefer if you did a 4 MB version instead of the 1 MB, it gives us a little bit more time when testing.

**Comment by Tyestor** [comment]
@Nadahar Refresh! :P - I'll be able to test the file in an hour or so. And yeah, I can make it bigger.

**Comment by Nadahar** [comment]
@Tyestor I blame GitHub 😊 

In any case, the file should have the same issue but I'd like for you to confirm before we start working on it.

**Comment by Tyestor** [comment]
@Nadahar 

Okay so, changing 1024 to 4096 changed the filesize to 40mb from 10mb, I can no longer upload it GitHub.

**Comment by Nadahar** [comment]
@Tyestor My bad again :P Forget it, and use the 10 MB fiile. I was a bit quick at reading, @Sami32 has that covered. The size produced is actually 1024 * 10000 😊 

I blame ME 😉

**Comment by Sami32** [comment]
@Tyestor Thanks. Excatly what i was thinking, but that your MediaInfo log didn't show up.

Definitively parsing issues are on our side, not the first ones that said...

**Comment by Tyestor** [comment]
@Nadahar @Sami32 

Okay guys, I just tested the sample file and the sound error is reproduced in it. Also, a possible unrelated weirdness, the thumbnail in both the real file and the sample is garbled/pixelated. Another bug?

**Comment by Nadahar** [comment]
@Tyestor The thumbnail issue is probably unrelated, but I just tested on your sample file and my thumbnail looks just fine, but I use the latest (unreleased) version. The thumbnail support has been completely rewritten since 6.6.0, so that will likely "fix itself" with the next release.

**Comment by Sami32** [comment]
@Tyestor Could you try with this draft UMS version ? Thanks.
As i'm not familiar or owner of a PS3, i rely on you for the feedback.

https://we.tl/OvE3X4BBLO

**P.S.** If any issue arise, please, post a new trace log.

**Comment by Tyestor** [comment]
@Sami32 @Nadahar 

Hey guys, I am happy to report that that build works perfectly (as far as I can tell anyway). It even fixed the garbled thumbnails (on the problem file in question/the sample file, some other files still have bunk thumbnails). I tested with LPCM audio only selected, everything else unchecked and then I tested with all four settings checked, both times the file played perfectly, no 'unholy screech' =)

Here's the log files, for good measure: [DTS-debug-logs.zip](https://github.com/UniversalMediaServer/UniversalMediaServer/files/994085/DTS-debug-logs.zip)

When I get the time, I'm going to create an issue ticket and see if I can get my other problems resolved, you guys are great.

**Comment by Nadahar** [comment]
@Tyestor Credits go entirely to @Sami32 for this 😉

**Comment by Tyestor** [comment]
@Nadahar 

Okay :D

@Sami32 

There may be one problem file, I was going through a bunch of different things and playing them to see if they had audio still and came across one that doesn't after change to this build. I will post a sample file/logs/mediainfo in this post as soon as I'm done testing the rest of the files.

**Comment by Sami32** [comment]
@Tyestor Thank you to you to have taken the time to report and give log and sample file, that really helped a lot ;)

The images/thumbnails fixes come from @Nadahar and @valib work :wink: 

As i said, "a draft", some important points was not yet properly fixed in it, just dirtly patched for testing.
You should try that one before any new report:
https://we.tl/XA2BmhyRMd

**Comment by Tyestor** [comment]
@Sami32 

The new build doesn't fix File 2's problem, no regressions as far as I can tell.

**Comment by Sami32** [comment]
@Tyestor Maybe a video sample from file 2 ? :wink: 

Yes, the duration mismatch could be the culprit, but a sample should help us.
Before send us the sample, test it and verify that you still get the issue, as the mismatch will be no there anymore.

**Comment by Tyestor** [comment]
@Sami32 I already posted a sample ;)

I did some more testing, though. Unticking 'Keep DTS Tracks' fixes File 2 (audio plays properly), so I'm assuming it's being transcoded to something. Whereas unticking 'Keep AC-3 tracks' fixes File 1.

**Comment by Sami32** [comment]
@Tyestor Sorry, i have refreshing browsing issue ;) Thanks.

**Comment by Sami32** [comment]
@Tyestor Yes, you was right, that your issue come from, your second video file is truncated so the duration mismatch.
When transcoded, FFmpeg fix it on the way.

Right now, i don't really know how to handle broken file, i will think about. I can recognize them and make them automatically transcoded, but it should be as the last resort if i don't find a better way.

**Comment by Tyestor** [comment]
@Sami32 That's okay dude, do what you can =)

**Comment by Sami32** [comment]
@SubJunk Do automatic transcoding for broken / truncated files seem acceptable to you ?
Or do you have an idea how to handle it ?

**Comment by Nadahar** [comment]
@Sami32 Does the renderer refuse to play it simply because the size/duration is wrong? Most players handle truncated files relativly well in my experience.

**Comment by Sami32** [comment]
@Nadahar No.
The problem is that for _PlayStation_ renderers the audio is defered to _tsMuXeR_, and it's hardcoded.
If _tsMuXeR_ was not used for such case it will be fine, IMHO.

But if i touch to that part, i'll make a different PR, as it could be never accepted.
What i can do now, it's add an entry in the logger for such case, avoiding us to search for nothing, as you already know we're very lucky when users test or post their samples and MediaInfo log.

**Comment by Tyestor** [comment]
@Sami32 @Nadahar 

We can close this ticket as the problem I had initially has been resolved thanks to @Sami32's excellent work. You can also merge the open pull request that relates to this issue (if it's ready). I can open a new ticket and with the relevant information that relates to the new issue (the truncated/broken .mkv file), and you can create a new PR for that issue, if you wish.

**Comment by Sami32** [comment]
@Tyestor Could try this new version, and post a new log after having reset the cache and runned a library scan ? Thanks.
https://we.tl/Pl9UfUZGvQ

It has a better broken media files logger that will give me a "real" view of your media files status.
As you've need to cut it to be able to post it, it could have been altereate the true nature of your media.

I added it to the DTS-HD PR, but the merge of my very modest work doesn't depend on me, but on BigBrother :wink: 

**Note**: To produce a more verbose MediaInfo's log, you need to select "Advanced Mode" in "Debug" tab.

**Comment by Sami32** [comment]
@Tyestor If you're still interested, i did a quick fix for your AC3 issue with your PS3.
So let me know if you want try it, i'll post you a download link.

**Comment by Tyestor** [comment]
Hey sorry @Sami32, I've been sick the last week or so + I had family visiting. I haven't had the time to do any tests so I will get on that ASAP. Please post the new link (i'll do the old link too). 

Thanks mate!

**Comment by Sami32** [comment]
@Tyestor Here it is:
https://wetransfer.com/downloads/0f3b8c74718a23c6aa612f24329ca7eb20170521185652/d1394f817ceded6cc78ed1974a816c4b20170521185652/33f42d

Please, whatever will be your feedback, post your new log, thanks.

**Comment by Tyestor** [comment]
Hey @Sami32 

Got some time finally.

Here's the missing mediainfo files, advanced mode of course.

File 1: [no-audio-mediainfo.txt](https://github.com/UniversalMediaServer/UniversalMediaServer/files/1027951/no-audio-mediainfo.txt)

File 2: [unholy-screech-2-advanced-mediainfo.txt](https://github.com/UniversalMediaServer/UniversalMediaServer/files/1027952/unholy-screech-2-advanced-mediainfo.txt)

I will test the log files shortly, editing this post.

**Comment by Tyestor** [comment]
@Sami32 @Nadahar 

Okay, so, for File 1 (aka no-audio), your latest build seems to have fixed the incorrect detection of the audio stream, it now shows 'Dolby Digital 5.1' in the PS3 status bar instead of, IIRC, LPCM 2.0. However, I still have a problem with it. For every other file I have checked, even ones that have 5.1 audio streams, they display 2.0 in the status bar. 

This leads me to believe that UMS, or my TV, or SOMETHING, is detecting my audio settings on my PS3 (I have all 5.1/7.1 LPCM audio streams disabled as they cause missing voices/low voices) and is forcing the 5.1 files I play to be re-encoded. This does not seem to be the case with File 1 as I get no audio unless I enable 5.1 LPCM on my PS3. However, that is unacceptable because I would constantly have to enable/disable it after/before playing video games, watching dvds and other movies off my PC.

As for File 2 (aka unholy screech 2 aka truncated file), there has been no change to it, still making a weird noise. 

Oh and there is a new problem, the original movie (Jason Bourne) now no longer plays with the new build you gave me, I get a 'This content cannot be played (800288e1)' error.

Here's an advanced mediainfo file of the first movie: [first-movie-advanced-mediainfo.txt](https://github.com/UniversalMediaServer/UniversalMediaServer/files/1028097/first-movie-advanced-mediainfo.txt)

I am doing the UMS logs now with build 'ums-PS3_AC3_transcoding_issue-8a2cfd6.exe'.

JB:
HP1: 
KoH:

**Comment by Sami32** [comment]
@Tyestor Sorry, i guess that i was not clear for the _MediaInfo_ log :/
![capture_05252017_143950](https://cloud.githubusercontent.com/assets/18471069/26450389/7237cd52-4157-11e7-8990-dbb61aff3de7.jpg)

File 2: Normal, as nothing was done, as i still don't have the _MediaInfo_ log that i asked.
I cannot fix it if i don't understand what happen. That's said, it's only one file ;)

> For every other file I have checked, even ones that have 5.1 audio streams, they display 2.0 in the status bar.

As they need to be transcoded or remuxed on PS3 renderer, it use your AC-3 2 channels user's setting. That's perfectly fine, so if it doesn't work same with your File 1, that mean that i missed or forgotten something.
I'll check at your log. Thanks for report ;)

**Comment by Tyestor** [comment]
Hey @Sami32 

Sorry about that.

HP1: [HP1-mediainfo-advanced-d0.txt](https://github.com/UniversalMediaServer/UniversalMediaServer/files/1030737/HP1-mediainfo-advanced-d0.txt)
KoH: [KoH-mediainfo-advanced-d0.txt](https://github.com/UniversalMediaServer/UniversalMediaServer/files/1030738/KoH-mediainfo-advanced-d0.txt)

**Comment by Sami32** [comment]
@Tyestor Thanks :+1: 

When you play _Kingdom.of.Heaven.2005.3in1.Ultimate.Edition.1080p.BluRay.x264-CtrlHD.mkv_ with VLC, or any other that you'll named, what is the duration displayed ?

**Comment by Sami32** [comment]
@Tyestor OK, i think i get your AC-3 issue, it was a multiple bugs issue ;)
But you'll need to joint an other log with a video file (AC-3) that work and Harry Potter with and without "AC-3 remux option". Thanks for your understanding.

As a workaround for now, i think that if you use AviSynth/FFmpeg from the #--TRANSCODE--# folder, your Harry Potter movie should play AC-3 fine.

**Comment by Tyestor** [comment]
@Sami32 

So if I'm understanding correctly you need me to

1. Check if KoH has the correct/incorrect time in MPC-HC (this is what I use), and if it has the correct time, open a new issue with the UMS logs, mediainfo, sample file and other relevant information.

2. Post logs, mediainfo and sample from another video file that has AC-3 audio that **works** on my PS3. This might be difficult, but I have a pretty large collection, so I will nonetheless look.

3. Post logs, mediainfo and sample from another video file that has DTS audio **NOT** DTS-HD.

4. Disable MPlayer thumbnails in the stable version of 7.0 :P

Okay so, as far as I can tell, my TV (Toshiba Regza) is stereo. It's pretty old, but it does have a number of HDMI ports, it's 1080p and it does have some Dolby tech in it (it just says Dolby Digital/Dolby Volume on the front). It also has surround virtualisation (the menu labels these settings completely uselessly as 'cinema' and ....something else I completely forgot, but it's useless, I assure you). 

I have no idea where the manual/specsheet is so I can't be sure what is actually in the thing and I can't find the real model number on it anywhere so even though I found a Regza manual online, it's a manual for multiple models. Even so, the specsheet I found was also utterly useless. So now I'm operating under the assumption that my TV is stereo. Should I not be? Should I change the AC-3 re-encoding option back to 5.1?

**Comment by Sami32** [comment]
@Tyestor 
1) If the time indicated by MPC-HC when playing  KoH is not the same as the MediaInfo duration, report the issue at the MediaInfo link given above.
2) Yes, but sample should not be need. From what i saw you've some 50 movies.
3) yes, from a DTS that work, if you ever have one, but sample should not be need.
4) No. Just use MEncoder to play the movie, and report log if you got audio playing correctly.

Go in "Transcoding Settings" tab, then in "Audio settings" sub tab, and unselect "Keep Ac-3 tracks" and all the others audio options, while having your maximum channels set to 2.
Then all your movies should work as expected, if not select "Encoded audio passthrough for AC-3 and DTS".

**Comment by Sami32** [comment]
@Tyestor I think that i have finally tackled your issues...
Could you give a try to that buid and give feedback ? (need a clean install to have get effect)
https://we.tl/lCggUKiLxm (this link is valid for 7 days)

Thank you for your help :+1: 

**Note**: As it is a testing build, be aware that the subtitles are totally disabled.
If you get an incorrect aspect ratio, just add this line into your renderer's configuration file:
`KeepAspectRatio = true`

**Comment by Sami32** [comment]
@Tyestor Sorry, i should have said that you will need to use that renderer: (replace the existing one)
https://raw.githubusercontent.com/UniversalMediaServer/UniversalMediaServer/master/src/main/external-resources/renderers/Sony-PlayStation3.conf

**Comment by Sami32** [comment]
@Tyestor Well, i've done 2 more fixes since then, so that time should be the good one ;)
I've already included your renderer.
https://we.tl/Rd9xR1BB1s (link valid for 7 days, will work correctly only with a clean install)

**Note**: In case you don't use the latest firmware, go in the "Transcode Settings" tab then in the "Audio settings" sub tab. From there, unselect "Keep AAC-LC tracks".
